doc/source/tutorial/stats/sampling_tdr.rst

```diff
@@ -39,35 +39,112 @@ This can be changed by passing a ``variant`` parameter.
 An example of using this method is shown below:
 
     >>> from scipy.stats import TransformedDensityRejection
+    >>> from scipy.stats import norm
     >>> 
-    >>> class Distribution:
+    >>> class StandardNormal:
     ...     def pdf(self, x):
-    ...         if abs(x) <= 1:
-    ...             return 1 - x*x
-    ...         return 0
+    ...         # note that the normalization constant is not required
+    ...         return np.exp(-0.5 * x*x)
     ...     def dpdf(self, x):
-    ...         if abs(x) <= 1:
-    ...             return -2*x
-    ...         return 0
+    ...         return -x * np.exp(-0.5 * x*x)
     ... 
-    >>> dist = Distribution()
+    >>> dist = StandardNormal()
     >>> 
     >>> urng = np.random.default_rng(0xbb51a838c8a087854ad3643b2b268d43)
     >>> rng = TransformedDensityRejection(dist, seed=urng)
     >>> rng.rvs()
-    0.39606414669189344
+    0.7771420061983989
+
+In the above example, we have used the TDR method to sample from the standard
+normal distribution. Note that we can drop the normalization constant while
+computing the PDF. This usually helps speed up the sampling stage. Also, note
+that the PDF doesn't need to be vectorized. It should accept and return a
+scalar.
 
 It is also possible to evaluate the inverse of the CDF of the hat distribution
 directly using the ``ppf_hat`` method.
 
     >>> rng.ppf_hat(0.5)
-    -5.7844272705533106e-05
+    -0.00018050266342362759
+    >>> norm.ppf(0.5)
+    0.0
     >>> u = np.linspace(0, 1, num=10)
     >>> rng.ppf_hat(u)
-    array([       -inf, -0.58619335, -0.39060804, -0.22634331, -0.07436531,
-            0.07424874,  0.22622123,  0.39047127,  0.58601716,         inf])
-
-For densities with modes not close to 0 it is suggested to set either the
+    array([       -inf, -1.22227372, -0.7656556 , -0.43135731, -0.14002921,
+            0.13966423,  0.43096141,  0.76517113,  1.22185606,         inf])
+    >>> norm.ppf(u)
+    array([       -inf, -1.22064035, -0.76470967, -0.4307273 , -0.1397103 ,
+            0.1397103 ,  0.4307273 ,  0.76470967,  1.22064035,         inf])
+
+Apart from the PPF method, other attributes can be accessed
+to see how well the generator fits the given distribution. These are:
+
+* 'sqhratio': (area below squeeze) / (area below hat) for the generator. It
+  is a number between 0 and 1. Closer to 1 means that the hat and the squeeze
+  functions tightly envelop the distribution and fewer PDF evaluations are
+  required to generate samples. By default, it is kept above 0.99 but that
```

Comment:
... and fewer PDF evaluations are required to generate samples: the expected number of evaluations of the density is bounded by (1/sqhratio) - 1.

Reply:
Added.